lib/vector-core/src/config/global_options.rs

```diff
@@ -230,6 +237,21 @@ impl GlobalOptions {
         let mut telemetry = self.telemetry.clone();
         telemetry.merge(&with.telemetry);
 
+        // TODO: maybe do some better merging here
+        let mut merged_expire_metrics_per_metric_set = None;
+        if self.expire_metrics_per_metric_set.is_some()
+            || with.expire_metrics_per_metric_set.is_some()
+        {
+            merged_expire_metrics_per_metric_set = Some(
+                self.expire_metrics_per_metric_set
+                    .iter()
+                    .flatten()
+                    .cloned()
+                    .chain(with.expire_metrics_per_metric_set.iter().flatten().cloned())
+                    .collect(),
+            );
```

Comment:
I think this is equivalent:
```suggestion
        let merged_expire_metrics_per_metric_set = match (
            &self.expire_metrics_per_metric_set,
            &with.expire_metrics_per_metric_set,
        ) {
            (Some(a), Some(b)) => Some(a.iter().chain(b).cloned().collect()),
            (Some(a), None) => Some(a.clone()),
            (None, Some(b)) => Some(b.clone()),
            (None, None) => None,
        };
```